Kernel/Arch/x86/Processor.h

```diff
@@ -398,7 +398,7 @@ class Processor {
     NEVER_INLINE void switch_context(Thread*& from_thread, Thread*& to_thread);
     [[noreturn]] static void assume_context(Thread& thread, FlatPtr flags);
     FlatPtr init_context(Thread& thread, bool leave_crit);
-    static Vector<FlatPtr> capture_stack_trace(Thread& thread, size_t max_frames = 0);
+    static Vector<FlatPtr, 32> capture_stack_trace(Thread& thread, size_t max_frames = 0);
```

Comment:
That's a lot of stack space, but if the optimization is worth it, sure

Reply:
it's just 256 bytes tops, I don't think that counts as "a lot of stack space"...

Reply:
We were doing it anyways, just throwing it away at the end.